DatadogSessionReplay/Sources/Feature/RequestBuilders/ResourceRequestBuilder.swift

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import Foundation
+import DatadogInternal
+
+internal struct ResourceRequestBuilder: FeatureRequestBuilder {
+    /// Custom URL for uploading data to.
+    let customUploadURL: URL?
+    /// Sends telemetry through sdk core.
+    let telemetry: Telemetry
+    /// Builds multipart form for request's body.
+    var multipartBuilder: MultipartFormDataBuilder = MultipartFormData(boundary: UUID())
+
+    func request(for events: [Event], with context: DatadogContext) throws -> URLRequest {
+        let decoder = JSONDecoder()
+        let resources = try events.map { event in
+            try decoder.decode(CodableResource.self, from: event.data)
+        }
+        return try createRequest(resources: resources, context: context)
+    }
+
+    private func createRequest(resources: [CodableResource], context: DatadogContext) throws -> URLRequest {
+        var multipart = multipartBuilder
```

Comment:
Does the assignment to the local here perform a copy, or does this method have a side effect of modifying `multipartBuilder`?